herd/AArch64Sem.ml

```diff
@@ -2820,127 +2826,153 @@ module Make
         let ops = List.mapi op rlist in
         List.fold_right (>>::) ops (M.unitT [[()]])
 
+      (** branch on whether [p]'s value [pred] has any active elements,
+          add [iico_causality_ctrl] to [mtrue] or [mfalse] *)
+      let any_active p pred psize nelem ii mtrue mfalse =
+        let>= any = get_predicate_any pred psize nelem in
+        let>* () =
+          let cond = Printf.sprintf "AnyActive(%s)" (A.pp_reg p) in
+          commit_pred_txt (Some cond) ii
+        in
+        M.choiceT any mtrue mfalse
+
+      (** branch on whether [idx] has past the last active predicate,
```

Comment:
Is this meant to say passed perhaps? Or maybe it would be simpler to say something along the lines of
```
check the element [idx] in predicate [pred] and add [mtrue] if active, or [mfalse] otherwise, ...
```